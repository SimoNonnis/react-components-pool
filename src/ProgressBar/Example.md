Progress bar with only required props

```jsx
<ProgressBar amount={10} />
```

Progress bar with lightBlue color

```jsx
<ProgressBar amount={40} color="lightBlue2" />
```

Progress bar with orange color showing the amount been passed in %

```jsx
<ProgressBar amount={50} color="orange" showValue />
```

Progress bar with green color showing the raw amount been passed

```jsx
<ProgressBar amount={2} color="green" showRawValue max={3} />
```
