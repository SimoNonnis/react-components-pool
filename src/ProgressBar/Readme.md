ProgressBar fill from 0% to 100% to show the progress of a task a for example.
The only required prop is amount:

```jsx
<ProgressBar amount={10} />
```

Progress bar with lightBlue color

```jsx
<ProgressBar amount={40} color="lightBlue2" />
```

Progress bar with orange color showing the amount been passed in %

```jsx
<ProgressBar amount={50} color="orange" showValue />
```

Progress bar with green color showing the raw amount been passed

```jsx
<ProgressBar amount={2} color="green" showRawValue max={3} />
```
