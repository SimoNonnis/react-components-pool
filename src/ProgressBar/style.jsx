import styled from 'styled-components';

export const Container = styled.div`
  align-items: stretch;
  background-color: ${({ theme }) => theme.color.lightGrey3};
  border-radius: 20px;
  display: flex;
  min-height: 15px;
`;
Container.displayName = 'Container';

export const Bar = styled.span`
  align-items: center;
  background-color: ${({ theme, color }) => theme.color[color]};
  border-radius: 20px;
  color: ${({ theme }) => theme.color.white};
  display: flex;
  font-family: ${({ theme }) => theme.base.fontFamily};
  font-size: ${({ theme }) => theme.base.fontSize};
  justify-content: center;
  min-height: 15px;
  transition: all 0.25s ease-in;
`;
Bar.displayName = 'Bar';
