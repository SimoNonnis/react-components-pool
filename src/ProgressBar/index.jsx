import React from 'react';
import PropTypes from 'prop-types';

import { Container, Bar } from './style';

const propTypes = {
  amount: PropTypes.number.isRequired,
  max: PropTypes.number,
  color: PropTypes.string,
  className: PropTypes.string,
  showValue: PropTypes.bool,
  showRawValue: PropTypes.bool,
};

const defaultProps = {
  max: 100,
  color: 'mediumBlue',
  showValue: false,
  showRawValue: false,
  className: '',
};

function ProgressBar({ max, amount, color, showValue, showRawValue, className }) {
  const barStyle = {
    width: `${amount / max * 100}%`,
  };
  let value;

  if (showValue && !showRawValue) {
    value = `${Math.ceil(amount / max * 100)}%`;
  } else if (showRawValue && !showValue) {
    value = amount;
  } else {
    value = null;
  }

  return (
    <div className={className}>
      <Container color={color}>
        <Bar color={color} style={barStyle}>
          {value}
        </Bar>
      </Container>
    </div>
  );
}

ProgressBar.propTypes = propTypes;
ProgressBar.defaultProps = defaultProps;

export default ProgressBar;
