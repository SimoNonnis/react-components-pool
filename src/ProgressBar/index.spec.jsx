/* eslint-env jest */
import React from 'react';

import ProgressBar from './';

describe('Test <ProgressBar />', () => {
  it('should render ProgressBar correctly passing only required props', () => {
    const wrapper = shallow(<ProgressBar amount={90} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render ProgressBar correctly with color prop', () => {
    const wrapper = shallow(<ProgressBar amount={90} color="green" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Progress bar with showValue prop', () => {
    const wrapper = shallow(<ProgressBar amount={90} showValue />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Progress bar with max value', () => {
    const wrapper = shallow(<ProgressBar amount={2} max={3} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Progress bar with max value and showRawValue prop', () => {
    const wrapper = shallow(<ProgressBar amount={2} max={3} showRawValue />);
    expect(wrapper).toMatchSnapshot();
  });
});
