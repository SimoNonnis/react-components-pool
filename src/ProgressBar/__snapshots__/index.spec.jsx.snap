// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <ProgressBar /> should render Progress bar with max value 1`] = `
<div
  className=""
>
  <Container
    color="mediumBlue"
  >
    <Bar
      color="mediumBlue"
      style={
        Object {
          "width": "66.66666666666666%",
        }
      }
    />
  </Container>
</div>
`;

exports[`Test <ProgressBar /> should render Progress bar with max value and showRawValue prop 1`] = `
<div
  className=""
>
  <Container
    color="mediumBlue"
  >
    <Bar
      color="mediumBlue"
      style={
        Object {
          "width": "66.66666666666666%",
        }
      }
    >
      2
    </Bar>
  </Container>
</div>
`;

exports[`Test <ProgressBar /> should render Progress bar with showValue prop 1`] = `
<div
  className=""
>
  <Container
    color="mediumBlue"
  >
    <Bar
      color="mediumBlue"
      style={
        Object {
          "width": "90%",
        }
      }
    >
      90%
    </Bar>
  </Container>
</div>
`;

exports[`Test <ProgressBar /> should render ProgressBar correctly passing only required props 1`] = `
<div
  className=""
>
  <Container
    color="mediumBlue"
  >
    <Bar
      color="mediumBlue"
      style={
        Object {
          "width": "90%",
        }
      }
    />
  </Container>
</div>
`;

exports[`Test <ProgressBar /> should render ProgressBar correctly with color prop 1`] = `
<div
  className=""
>
  <Container
    color="green"
  >
    <Bar
      color="green"
      style={
        Object {
          "width": "90%",
        }
      }
    />
  </Container>
</div>
`;
